contracts/Staking.sol

```diff
@@ -352,7 +357,19 @@ contract Staking is Ownable {
         uint256 burnedDUTokens = _amount.mul(p0).div(t0);
         duToken.burnFrom(msg.sender, burnedDUTokens);
         usersPool[_tokenAddress] = usersPool[_tokenAddress].sub(_amount);
-        ERC20(_tokenAddress).safeTransfer(msg.sender, _amount);
+        uint outstandingAmount = _distributeClaimingRewards(_amount, _tokenAddress);
+        ERC20(_tokenAddress).safeTransfer(msg.sender, outstandingAmount);
+    }
+
+    function _distributeClaimingRewards(
+        uint _amount,
+        address _tokenAddress
+    ) internal returns (uint256) {
+        uint slaOwnerRewards = _amount.mul(ownerRewardsRate).div(10000);
+        uint protocolRewards = _amount.mul(protocolRewardsRate).div(10000);
+        ERC20(_tokenAddress).safeTransfer(owner(), slaOwnerRewards);
+        ERC20(_tokenAddress).safeTransfer(_stakeRegistry.owner(), protocolRewards);
+        return _amount.div(slaOwnerRewards).div(protocolRewards);
```

Comment:
Doesn't look like this is returning the outstanding amount.
Change to `_amount.sub(slaOwnerRewards).sub(protocolRewards)`